mongo/bson.nim

```diff
@@ -809,17 +831,19 @@ proc to*(b: Bson, T: typedesc): T =
       var key = k
       when val.hasCustomPragma(dbKey):
         key = val.getCustomPragmaVal(dbKey)
-      if key notin b:
+      if key notin b and not val.hasCustomPragma(omitempty):
         raise BsonError.newException "Key " & key & " not found for " & $T
-      val = b[key].to(type(val))
+      if key in b:
+        val = b[key].to(type(val))
   elif T is object|tuple:
     for k, val in fieldPairs(result):
       var key = k
       when val.hasCustomPragma(dbKey):
         key = val.getCustomPragmaVal(dbKey)
-      if key notin b:
+      if key notin b and not val.hasCustomPragma(omitempty):
         raise BsonError.newException "Key " & key & " not found for " & $T
-      val = b[key].to(type(val))
+      if key in b:
+        val = b[key].to(type(val))
```

Comment:
Please update uses of `type` to modern `typeof`; the former is deprecated for this purpose.